### DISCUSSION TITLE
Store with dynamic keys/attributes

### DISCUSSION THREAD

**Discussion by MHC03** [pmndrs/zustand#1213] [Q&A] [answered]
Hi everyone,

first and foremost, thank you very much for developing this beautiful state-management library. I really like it!

I would like to know, if there are any ramifications if you set the keys/attributes dynamically in the store. Let's take the following example:

```ts
type Store = {
[id: string]: number;
}

const initialState: Store = {};

export const useDataStore = create<Store>()(_ => initialState);

export const addData = (someID: string, someNumber: number): void => {
  useDataStore.setState(state => ({...state, [someID]: someNumber}));
}

export const useSomeData = (someID: string): number => {
  return useDataStore(state => state[someID]) ?? 0;
}
```

Is there any downside to thist method? The alternative would be to define my store with one attribute:

```ts
type StoreWithAttribute = {
  data: {[id: string]: number};
}
```
I thought this would be unnecessary, but would everything work normally if following the first code example?

**Comment by dai-shi** [comment, marked as answer]
Sure it works. It's just a little typescript unfriendly. It has nothing to do with zustand. dynamic keys are not friendly to static typing.

**Reply by MHC03** [reply]
Thank you very much! In this case the types  won't make any difference, because, I would have used the same dynamic structure, just with an extra (unnecessary) attribute.
